spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/ExpressionEvaluatingSqlParameterSourceFactory.java

```diff
@@ -126,9 +129,23 @@ public void setParameterExpressions(Map<String, String> parameterExpressions) {
 		this.parameterExpressions = paramExpressions;
 	}
 
+	/**
+	 * Optionally set parameter types for the used parameters. Use
+	 * {@link java.sql.Types} to get the parameter type value.
+	 * @param sqlParametersTypes the parameter type to set
+	 * @see java.sql.Types
+	 * @since 5.0
+	 */
+	public void setSqlParameterTypes(Map<String, Integer> sqlParametersTypes) {
+		this.sqlParametersTypes = sqlParametersTypes;
+	}
+
 	@Override
 	public SqlParameterSource createParameterSource(final Object input) {
-		return new ExpressionEvaluatingSqlParameterSource(input, this.staticParameters, this.parameterExpressions, true);
+		AbstractSqlParameterSource sqlParameterSource = new ExpressionEvaluatingSqlParameterSource(input,
+				this.staticParameters, this.parameterExpressions, true);
+		enrichSqlParametersType(sqlParameterSource);
```

Comment:
I would say `registerSqlTypes` to reflect an existing `AbstractSqlParameterSource.registerSqlType()`

Reply:
I'll make it `registerSqlTypes()`.